workflows/shared.py

```diff
@@ -107,34 +113,28 @@ def free_port_selector(node_subscription_id: UUIDstr, speed: int, enum: str = "P
     return Choice(enum, zip(interfaces.keys(), interfaces.items()))  # type:ignore
 
 
-def summary_form(product_name: str, summary_data: dict) -> Generator:
-    class ProductSummary(MigrationSummary):
-        data = SummaryData(**summary_data)
+def summary_form(product_name: str, summary_data: SummaryData) -> Generator:
+    ProductSummary: TypeAlias = cast(type[MigrationSummary], migration_summary(summary_data))
 
     class SummaryForm(FormPage):
-        class Config:
-            title = f"{product_name} summary"
+        model_config = ConfigDict(title=f"{product_name} summary")
 
-        product_summary: ProductSummary
+        product_summary: ProductSummary  # type: ignore
```

Comment:
If the typealias isn't working, might as well not use a typealias. I'll try running mypy to see what the problem is

Reply:
The type ignore can be removed